packages/vite/src/node/plugins/html.ts

```diff
@@ -543,42 +587,20 @@ export function buildHtmlPlugin(config: ResolvedConfig): Plugin {
           )
         }
 
-        // for each encountered asset url, rewrite original html so that it
-        // references the post-build location, ignoring empty attributes and
-        // attributes that directly reference named output.
-        const namedOutput = Object.keys(
-          config?.build?.rollupOptions?.input || {},
-        )
-        for (const { attr, sourceCodeLocation } of assetUrls) {
-          // assetsUrl may be encodeURI
-          const content = decodeURI(attr.value)
-          if (
-            content !== '' && // Empty attribute
-            !namedOutput.includes(content) && // Direct reference to named output
-            !namedOutput.includes(removeLeadingSlash(content)) // Allow for absolute references as named output can't be an absolute path
-          ) {
-            try {
-              const url =
-                attr.prefix === undefined && attr.name === 'srcset'
-                  ? await processSrcSet(content, ({ url }) =>
-                      urlToBuiltUrl(url, id, config, this),
-                    )
-                  : await urlToBuiltUrl(content, id, config, this)
-
-              overwriteAttrValue(s, sourceCodeLocation, url)
-            } catch (e) {
-              if (e.code !== 'ENOENT') {
-                throw e
-              }
-            }
+        try {
+          await Promise.all(assetUrlsPromises)
+        } catch (e) {
+          if (e.code !== 'ENOENT') {
+            throw e
```

Comment:
I think we should move this `catch` into `processAssetUrl` so that we ignore errors properly.
https://github.com/vitejs/vite/commit/611fa037a72a1179c27794353ffad6ed27e10d1a

Reply:
Done at https://github.com/vitejs/vite/pull/14888/commits/6414fb32b3a7d9e79f84e11dac1db96f05d597a3